BUILD

```diff
@@ -1467,10 +1467,10 @@ grpc_cc_library(
 )
 
 grpc_cc_library(
-    name = "xds_channel_creds",
+    name = "channel_creds",
```

Comment:
Suggest calling this `channel_creds_registry`.

Reply:
Done.

> This looks pretty good! Just a few organizational comments remaining.
> 
> For future reference, please do not rebase or force-push in the middle of a review, since that makes it very hard for the reviewer to track the changes. If you really need to merge in changes from master in the middle of a review, just merge from master into your branch and push that merge to github.
> 
> Thanks!

Sure, I had to merge #28819 due to failing tests. So I guess the merge commit created after a rebase will go away when I squash and commit later, will avoid force-pushing definitely.